test/plugins/WebOfTrust/util/TickerDelayedBackgroundJobTest.java

```diff
@@ -0,0 +1,393 @@
+     * when receiving an InterruptedException during sleep.
+     * @param sleepTime the sleep time in ms
+     */
+    private Runnable newValueIncrementer(final long sleepTime) {
+        return new Runnable() {
+            private AtomicBoolean isRunning = new AtomicBoolean(false);
+            @Override
+            public void run() {
+                if (!isRunning.compareAndSet(false, true)) {
+                    wasConcurrent.set(true);
+                }
+                value.incrementAndGet();
+                try {
+                    Thread.sleep(sleepTime);
+                } catch (InterruptedException e) {
+                    wasInterrupted.set(true);
+                }
+                isRunning.set(false);
+            }
+        };
+    }
+
+    /**
+     * Creates a Runnable that invokes {@code job.triggerExecution()} 1000 times, sleeps for about
+     * 1 ms, and repeats this for {@code duration} ms.
+     * The created Runnable is stateless and can be used multiple times, even concurrently.
+     */
+    private Runnable newHammerDefault(final DelayedBackgroundJob job, final long duration) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                long t = System.currentTimeMillis();
+                while (System.currentTimeMillis() < t + duration) {
+                    for (int i = 0; i < 1000; i++) {
+                        job.triggerExecution();
+                    }
+                    try {
+                        Thread.sleep(1, rand.addAndGet(500));
+                    } catch (InterruptedException e) {
+                        wasInterrupted.set(true);
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * Creates a Runnable that invokes {@code job.triggerExecution(long)} 1 time, sleeps for about
+     * 1 ms, and repeats this until all delays are used (first to last).
+     * The created Runnable is stateless and can be used multiple times, even concurrently.
+     */
+    private Runnable newHammerCustom(final DelayedBackgroundJob job, final long[] delays) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                for (long delay : delays) {
+                    job.triggerExecution(delay);
+                    try {
+                        Thread.sleep(1, rand.addAndGet(500));
+                    } catch (InterruptedException e) {
+                        wasInterrupted.set(true);
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * Warmup the job/ticker/executor by triggering its immediate execution and waiting for the
+     * change to happen 10 times. Restores the {@link #value} after warmup.
+     * @param job the job to warm up
+     * @param jobDuration the expected duration of the job
+     */
+    private void warmup(TickerDelayedBackgroundJob job, long jobDuration) throws Exception {
```

Comment:
OK so it is to trigger JIT compilation, which I had also assumed :)
FYI: I have rewritten this, mostly because it didn't cause sufficient JIT for me. I googled the magical numbers and it seems that by default, Java will compile a function after 1500 executions (or 1501, dunno whether its inclusive).